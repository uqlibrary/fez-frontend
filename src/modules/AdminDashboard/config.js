import React from 'react';
import moment from 'moment';

import Chip from '@mui/material/Chip';
import Typography from '@mui/material/Typography';

import { ExternalLink } from 'modules/SharedComponents/ExternalLink';

export const COLOURS = { assigned: '#338CFA', unassigned: '#B60DCE' };

export const LINK_UNPROCESSED_WORKS =
    'https://espace.library.uq.edu.au/records/search?page=1&pageSize=20&sortBy=score&sortDirection=Desc&searchQueryParams%5Ball%5D%5Bvalue%5D=ismemberof%3A+UQ%3A218198&searchQueryParams%5Ball%5D%5Blabel%5D=&searchMode=advanced';

export const animationTemplate = (i, duration, delay) =>
    `animateFadeIn ${duration}ms ease-out ${delay * (i + 1)}ms forwards`;

export const MENUACTIONS = {
    EDIT: 'EDIT',
    DELETE: 'DELETE',
    UP: 'MOVEUP',
    TOP: 'MOVETOP',
    DOWN: 'MOVEDOWN',
    BOTTOM: 'MOVEBOTTOM',
};

export const VIEWMODES = {
    VIEW: 'VIEW',
    ADD: 'ADD',
    EDIT: 'EDIT',
    DELETE: 'DELETE',
};
export const VIEWADMINPANELMODES = [VIEWMODES.ADD, VIEWMODES.EDIT, VIEWMODES.DELETE];
export const REORDERING = [MENUACTIONS.TOP, MENUACTIONS.UP, MENUACTIONS.BOTTOM, MENUACTIONS.DOWN];

export const DEFAULT_DATEPICKER_INPUT_FORMAT = 'DD/MM/YYYY';
export const DEFAULT_DATE_FORMAT = 'Do MMMM YYYY';
export const DEFAULT_DATE_FORMAT_WITH_TIME = 'Do MMMM YYYY hh:mm';
export const DEFAULT_DATE_FORMAT_WITH_TIME_24H = 'Do MMMM YYYY HH:mm';
export const DEFAULT_DATE_FORMAT_WITH_TIME_24H_SECONDS = 'Do MMMM YYYY HH:mm:ss';
export const DEFAULT_SERVER_DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss';

export const SYSTEM_ALERT_ACTION = {
    ASSIGN: 'ASSIGN',
    RESOLVE: 'RESOLVE',
};

export const REPORT_TYPE = {
    systemalertlog: 1,
    workshistory: 2,
    workiddupe: 3,
    scopusiddupe: 4,
    doidupe: 5,
};

export const DEFAULT_SORTING = {
    alerts: [{ field: 'sat_created_date', sort: 'asc' }],
    systemalertlog: [{ field: 'sat_created_date', sort: 'asc' }],
    workshistory: [{ field: 'pre_date', sort: 'asc' }],
};

export const isUrl = str => {
    try {
        const newUrl = new URL(str);
        return newUrl.protocol === 'http:' || newUrl.protocol === 'https:';
    } catch (err) {
        return false;
    }
};

export const defaultLegacyReportOption = { sel_id: 0, sel_title: '', sel_description: '' };

export const optionDoubleRowRender = (props, option) => {
    return (
        <li
            {...props}
            style={{
                flexDirection: 'column',
                justifyContent: 'left',
                alignItems: 'flex-start',
                fontWeight: 400,
            }}
            data-testid={props.id}
        >
            <Typography variant="body1" color="textPrimary">
                {option.sel_title}
            </Typography>
            <Typography variant="body1" color="textSecondary">
                {option.sel_description}
            </Typography>
        </li>
    );
};

export const getReportTypeFromValue = value => Object.entries(REPORT_TYPE).find(arr => arr[1] === value)?.[0];
export const getDefaultSorting = reportType => DEFAULT_SORTING?.[reportType] || [];

export const getFormattedServerDate = (dateStr, format = DEFAULT_DATE_FORMAT) =>
    (dateStr && moment(dateStr, DEFAULT_SERVER_DATE_FORMAT).format(format)) || '';

export const getSystemAlertColumns = (locale, users) => {
    const alertStatus = locale.alertStatus;
    const alertStatusOption = Object.values(alertStatus);
    return [
        {
            field: 'sat_created_date',
            headerName: locale.columns.createdDate,
            width: 200,
            type: 'date',
            valueGetter: value => moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate(),
            valueFormatter: value => moment(value).format(DEFAULT_DATE_FORMAT_WITH_TIME_24H),
        },
        { field: 'sat_title', headerName: locale.columns.topic, flex: 1 },
        {
            field: 'status',
            headerName: locale.columns.status,
            width: 160,
            valueGetter: (_, row) =>
                !!row.sat_assigned_to
                    ? users.find(user => user.id === row.sat_assigned_to)?.preferred_name ?? alertStatus.UNKNOWN
                    : alertStatus.UNASSIGNED,
            renderCell: params => (
                <Chip
                    data-testid={`alert-status-${params.id}`}
                    label={params.value}
                    variant="filled"
                    size="small"
                    color={alertStatusOption.includes(params.value) ? 'default' : 'primary'}
                />
            ),
        },
    ];
};

export const getDisplayReportColumns = ({ locale, actionState, params }) => {
    const report = actionState?.displayReport?.value || getReportTypeFromValue(params.report_type);

    const txt = locale.columns[report];
    switch (report) {
        case 'workshistory':
            return [
                { field: 'pre_id', headerName: txt.id, width: 150, exportOnly: true, exportOrder: 0 },
                { field: 'pre_pid', headerName: txt.pid, width: 150, order: 0, exportOrder: 1 },
                {
                    field: 'pre_date',
                    headerName: txt.dateCreated,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 1,
                    exportOrder: 5,
                },
                {
                    field: 'rek_date',
                    headerName: txt.pubDate,
                    width: 150,
                    type: 'date',
                    valueGetter: value =>
                        (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).format(DEFAULT_DATE_FORMAT)) || '',
                    exportOnly: true,
                    exportOrder: 4,
                },
                { field: 'rek_genre', headerName: txt.genre, minWidth: 300, flex: 1, order: 2, exportOrder: 2 },
                { field: 'rek_subtype', headerName: txt.subtype, minWidth: 300, flex: 1, order: 3, exportOrder: 3 },
                { field: 'usr_username', headerName: txt.user, width: 150, order: 4, exportOrder: 6 },
                { field: 'pre_detail', headerName: txt.action, minWidth: 600, flex: 1, order: 5, exportOrder: 7 },
            ];
        default:
            const systemIdParam = actionState?.systemAlertId || params?.record_id || '';
            const cols = [
                { field: 'sat_id', headerName: txt.id, order: 0, exportOrder: 0 },
                {
                    field: 'sat_created_date',
                    headerName: txt.dateCreated,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 1,
                    exportOrder: 1,
                },
                {
                    field: 'assigned_to_username',
                    headerName: 'Assigned username',
                    width: 150,
                    exportOrder: 2,
                    exportOnly: true,
                },
                {
                    field: 'assigned_to_full_name',
                    headerName: txt.assignedTo,
                    width: 150,
                    exportOrder: 3,
                    order: 2,
                },
                {
                    field: 'sat_assigned_date',
                    headerName: txt.assignedDate,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 3,
                    exportOrder: 4,
                },
                {
                    field: 'resolved_by_username',
                    headerName: 'Resolved username',
                    width: 150,
                    exportOrder: 5,
                    exportOnly: true,
                },
                {
                    field: 'resolved_by_full_name',
                    headerName: txt.resolvedBy,
                    width: 150,
                    order: 4,
                    exportOrder: 6,
                },
                {
                    field: 'sat_resolved_date',
                    headerName: txt.resolvedDate,
                    width: 150,
                    type: 'date',
                    valueGetter: value => (!!value && moment(value, DEFAULT_SERVER_DATE_FORMAT).toDate()) || null,
                    valueFormatter: value => {
                        if (value === null || value === '') return '';
                        return moment(value).format(DEFAULT_DATE_FORMAT);
                    },
                    order: 5,
                    exportOrder: 7,
                },
                { field: 'sat_title', headerName: txt.title, minWidth: 400, flex: 1, order: 6, exportOrder: 8 },
                {
                    field: 'sat_link',
                    headerName: txt.link,
                    minWidth: 400,
                    flex: 1,
                    renderCell: row =>
                        isUrl(row.value) ? (
                            <ExternalLink id={`link_${row.id}`} href={row.value} inline>
                                {row.value}
                            </ExternalLink>
                        ) : (
                            row.value
                        ),
                    order: 8,
                    exportOrder: 9,
                },
            ];

            if (systemIdParam !== '') {
                cols.push({
                    field: 'sat_content',
                    headerName: txt.content,
                    minWidth: 1000,
                    flex: 1,
                    order: 7,
                    exportOrder: 10,
                });
            }
            return cols;
    }
};
