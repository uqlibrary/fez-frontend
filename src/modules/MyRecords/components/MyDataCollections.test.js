import MyDatasets from './MyRecords';
import { pathConfig } from 'config';
import { locale } from 'locale';
import React from 'react';
import { render, WithRouter, WithReduxStore } from 'test-utils';

function setup(testProps = {}, renderMethod = render) {
    const props = {
        actions: {
            loadAuthorPublications: jest.fn(),
            setFixRecord: jest.fn(),
        },
        location: {
            pathname: pathConfig.dataset.mine,
            state: null,
        },
        navigate: testProps.navigate || jest.fn(),
        navigationType: testProps.navigationType || '',
        accountLoading: false,
        localePages: locale.pages.myDatasets,
        publicationsListPagingData: {},
        loadingPublicationsList: false,
        publicationsList: [],
        publicationsListFacets: {},
        authorDetails: {
            is_administrator: 0,
            is_super_administrator: 0,
        },
        ...testProps,
    };
    return renderMethod(
        <WithReduxStore>
            <WithRouter>
                <MyDatasets {...props} />
            </WithRouter>
        </WithReduxStore>,
    );
}

describe('myDatasets test', () => {
    it('renders loading screen while loading account data', () => {
        const { container } = setup({ accountLoading: true });
        expect(container).toMatchSnapshot();
    });

    it('renders loading screen while loading publications ', () => {
        const container = setup({ loadingPublicationsList: true });
        expect(container).toMatchSnapshot();
    });

    it('renders loading screen while loading publications while filtering', () => {
        const { container } = setup({
            loadingPublicationsList: true,
            publicationsList: [{ rek_pid: 'UQ:1', rek_title: '1' }],
        });
        expect(container).toMatchSnapshot();
    });

    it('renders no results', () => {
        const { container } = setup();
        expect(container).toMatchSnapshot();
    });

    it('renders list of publications no facets', () => {
        const { container } = setup({
            publicationsList: [{ rek_pid: 'UQ:1', rek_title: '1' }], // myRecordsList.data,
            publicationsListPagingData: { total: 2, per_page: 20, current_page: 1, from: 1, to: 2 },
        });
        expect(container).toMatchSnapshot();
    });

    it('renders list of publications with facets', () => {
        const { container } = setup({
            publicationsList: [{ rek_pid: 'UQ:1', rek_title: '1' }], // myRecordsList.data,
            publicationsListPagingData: { total: 2, per_page: 20, current_page: 1, from: 1, to: 2 },
            publicationsListFacets: {
                'Display type': {
                    doc_count_error_upper_bound: 0,
                    sum_other_doc_count: 3,
                    buckets: [
                        { key: 179, doc_count: 95 },
                        { key: 130, doc_count: 34 },
                        {
                            key: 177,
                            doc_count: 2,
                        },
                        { key: 183, doc_count: 2 },
                        { key: 174, doc_count: 1 },
                    ],
                },
                Keywords: {
                    doc_count_error_upper_bound: 0,
                    sum_other_doc_count: 641,
                    buckets: [
                        { key: 'Brca1', doc_count: 15 },
                        {
                            key: 'Oncology',
                            doc_count: 15,
                        },
                        { key: 'Breast cancer', doc_count: 13 },
                        {
                            key: 'Genetics & Heredity',
                            doc_count: 12,
                        },
                        { key: 'Biochemistry & Molecular Biology', doc_count: 10 },
                    ],
                },
            },
        });
        expect(container).toMatchSnapshot();
    });

    it('renders active filters', () => {
        const { container } = setup({
            state: { activeFacets: { filters: {}, ranges: { Year: { from: 2000, to: 2010 } } } },
        });
        expect(container).toMatchSnapshot();
    });

    it('gets publications when user clicks back and state is set', () => {
        const testAction = jest.fn();
        const { container, rerender } = setup({
            accountLoading: true,
            actions: { loadAuthorPublications: testAction },
            thisUrl: pathConfig.dataset.mine,
        });

        setup(
            {
                navigationType: 'POP',
                location: { pathname: pathConfig.dataset.mine, state: { page: 2, hasPublications: true } },
            },
            rerender,
        );
        // expect(testAction).toBeCalled();
        expect(container).toMatchSnapshot();
    });

    it('gets publications when user clicks back and state is not set', () => {
        const testAction = jest.fn();
        const { container, rerender } = setup({
            accountLoading: true,
            actions: { loadAuthorPublications: testAction },
            thisUrl: pathConfig.dataset.mine,
        });

        setup(
            {
                navigationType: 'POP',
                location: { pathname: pathConfig.dataset.mine, state: null },
                loadingPublicationsList: false,
                publicationsList: [],
            },
            rerender,
        );
        // expect(testAction).toHaveBeenCalled();
        expect(container).toMatchSnapshot();
    });
});
