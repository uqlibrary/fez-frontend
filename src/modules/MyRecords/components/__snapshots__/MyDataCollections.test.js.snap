// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`myDatasets test renders active filters 1`] = `
<StandardPage
  title="My research data"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <StandardCard
        text={
          <div>
            We were unable to find any results. You may be able to
             
            <a
              href="/data-collections/add"
            >
              add a missing dataset
            </a>
            .
          </div>
        }
        title="No datasets found"
      >
        <div>
          We were unable to find any results. You may be able to
           
          <a
            href="/data-collections/add"
          >
            add a missing dataset
          </a>
          .
        </div>
      </StandardCard>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</StandardPage>
`;

exports[`myDatasets test renders list of publications no facets 1`] = `
<StandardPage
  title="My research data"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={9}
      xs={12}
    >
      <StandardCard
        noHeader={true}
      >
        <span>
          Displaying works 1 to 2 of 2 total works. 
        </span>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <span>
              Add to this list by 
              <a
                href="/data-collections/add"
              >
                adding a missing dataset
              </a>
              .
            </span>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <PublicationsListSorting
              bulkExportSize={1000}
              exportData={Object {}}
              initPageLength={20}
              onExportPublications={[Function]}
              onPageSizeChanged={[Function]}
              onSortByChanged={[Function]}
              pageSize={20}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 2,
                  "total": 2,
                }
              }
              showDisplayAs={false}
              sortBy="published_date"
              sortDirection="Desc"
              sortingData={
                Object {
                  "bulkExportConfirmation": Object {
                    "confirmButtonLabel": "Close",
                    "confirmationMessage": <span
                      style={
                        Object {
                          "display": "block",
                          "margin": "1em 0",
                        }
                      }
                    >
                      Bulk export request has been queued. When the request has been processed, you will receive an email with the exported file as an attachment.
                    </span>,
                    "confirmationTitle": "Export queued successfully",
                  },
                  "displayRecordsAs": Array [
                    Object {
                      "index": 0,
                      "label": "Standard",
                      "value": "standard",
                    },
                    Object {
                      "index": 1,
                      "label": "Image Gallery",
                      "value": "image-gallery",
                    },
                  ],
                  "displayRecordsAsLabel": "Display results",
                  "exportOnlyLabel": "Export Only:",
                  "pageSize": "Works per page",
                  "recordsPerPage": Array [
                    10,
                    20,
                    50,
                    100,
                  ],
                  "sortBy": Array [
                    Object {
                      "label": "Published date",
                      "value": "published_date",
                    },
                    Object {
                      "label": "Search relevance",
                      "value": "score",
                    },
                    Object {
                      "label": "Title",
                      "value": "title",
                    },
                    Object {
                      "label": "Created date",
                      "value": "created_date",
                    },
                    Object {
                      "label": "Updated date",
                      "value": "updated_date",
                    },
                    Object {
                      "exclude": Array [
                        "image-gallery",
                      ],
                      "label": "Altmetric score",
                      "value": "altmetric_score",
                    },
                    Object {
                      "exclude": Array [
                        "image-gallery",
                      ],
                      "label": "Scopus citation count",
                      "value": "scopus_citation_count",
                    },
                    Object {
                      "exclude": Array [
                        "image-gallery",
                      ],
                      "label": "Web of Science citation count",
                      "value": "thomson_citation_count",
                    },
                  ],
                  "sortDirection": Array [
                    "Desc",
                    "Asc",
                  ],
                  "sortDirectionLabel": "Sort order",
                  "sortLabel": "Sort results by",
                }
              }
              sortingDefaults={Object {}}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={false}
              onPageChanged={[Function]}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 2,
                  "total": 2,
                }
              }
              pagingId="my-records-paging-top"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(ConfirmDialogBox)
              hideCancelButton={true}
              locale={
                Object {
                  "confirmButtonLabel": "Close",
                  "confirmationMessage": <span
                    style={
                      Object {
                        "display": "block",
                        "margin": "1em 0",
                      }
                    }
                  >
                    Bulk export request has been queued. When the request has been processed, you will receive an email with the exported file as an attachment.
                  </span>,
                  "confirmationTitle": "Export queued successfully",
                }
              }
              onRef={[Function]}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <Memo(PublicationsList)
              publicationsList={
                Array [
                  1,
                  2,
                  3,
                ]
              }
              showAdminActions={false}
              showDefaultActions={true}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={false}
              onPageChanged={[Function]}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 2,
                  "total": 2,
                }
              }
              pagingId="my-records-paging-bottom"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </StandardCard>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</StandardPage>
`;

exports[`myDatasets test renders list of publications with facets 1`] = `
<StandardPage
  title="My research data"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={9}
      xs={12}
    >
      <StandardCard
        noHeader={true}
      >
        <span>
          Displaying works 1 to 2 of 2 total works. 
        </span>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <span>
              Add to this list by 
              <a
                href="/data-collections/add"
              >
                adding a missing dataset
              </a>
              .
            </span>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <PublicationsListSorting
              bulkExportSize={1000}
              exportData={Object {}}
              initPageLength={20}
              onExportPublications={[Function]}
              onPageSizeChanged={[Function]}
              onSortByChanged={[Function]}
              pageSize={20}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 2,
                  "total": 2,
                }
              }
              showDisplayAs={false}
              sortBy="published_date"
              sortDirection="Desc"
              sortingData={
                Object {
                  "bulkExportConfirmation": Object {
                    "confirmButtonLabel": "Close",
                    "confirmationMessage": <span
                      style={
                        Object {
                          "display": "block",
                          "margin": "1em 0",
                        }
                      }
                    >
                      Bulk export request has been queued. When the request has been processed, you will receive an email with the exported file as an attachment.
                    </span>,
                    "confirmationTitle": "Export queued successfully",
                  },
                  "displayRecordsAs": Array [
                    Object {
                      "index": 0,
                      "label": "Standard",
                      "value": "standard",
                    },
                    Object {
                      "index": 1,
                      "label": "Image Gallery",
                      "value": "image-gallery",
                    },
                  ],
                  "displayRecordsAsLabel": "Display results",
                  "exportOnlyLabel": "Export Only:",
                  "pageSize": "Works per page",
                  "recordsPerPage": Array [
                    10,
                    20,
                    50,
                    100,
                  ],
                  "sortBy": Array [
                    Object {
                      "label": "Published date",
                      "value": "published_date",
                    },
                    Object {
                      "label": "Search relevance",
                      "value": "score",
                    },
                    Object {
                      "label": "Title",
                      "value": "title",
                    },
                    Object {
                      "label": "Created date",
                      "value": "created_date",
                    },
                    Object {
                      "label": "Updated date",
                      "value": "updated_date",
                    },
                    Object {
                      "exclude": Array [
                        "image-gallery",
                      ],
                      "label": "Altmetric score",
                      "value": "altmetric_score",
                    },
                    Object {
                      "exclude": Array [
                        "image-gallery",
                      ],
                      "label": "Scopus citation count",
                      "value": "scopus_citation_count",
                    },
                    Object {
                      "exclude": Array [
                        "image-gallery",
                      ],
                      "label": "Web of Science citation count",
                      "value": "thomson_citation_count",
                    },
                  ],
                  "sortDirection": Array [
                    "Desc",
                    "Asc",
                  ],
                  "sortDirectionLabel": "Sort order",
                  "sortLabel": "Sort results by",
                }
              }
              sortingDefaults={Object {}}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={false}
              onPageChanged={[Function]}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 2,
                  "total": 2,
                }
              }
              pagingId="my-records-paging-top"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(ConfirmDialogBox)
              hideCancelButton={true}
              locale={
                Object {
                  "confirmButtonLabel": "Close",
                  "confirmationMessage": <span
                    style={
                      Object {
                        "display": "block",
                        "margin": "1em 0",
                      }
                    }
                  >
                    Bulk export request has been queued. When the request has been processed, you will receive an email with the exported file as an attachment.
                  </span>,
                  "confirmationTitle": "Export queued successfully",
                }
              }
              onRef={[Function]}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <Memo(PublicationsList)
              publicationsList={
                Array [
                  1,
                  2,
                  3,
                ]
              }
              showAdminActions={false}
              showDefaultActions={true}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={false}
              onPageChanged={[Function]}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 2,
                  "total": 2,
                }
              }
              pagingId="my-records-paging-bottom"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </StandardCard>
    </WithStyles(ForwardRef(Grid))>
    <Hidden
      smDown={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={3}
      >
        <WithStyles(StandardRighthandCard)
          title="Refine results"
        >
          <Memo(FacetsFilter)
            activeFacets={
              Object {
                "filters": Object {},
                "ranges": Object {},
              }
            }
            excludeFacetsList={
              Array [
                "Scopus document type",
                "Subtype",
                "Year published",
                "Display type",
              ]
            }
            facetsData={
              Object {
                "Display type": Object {
                  "buckets": Array [
                    Object {
                      "doc_count": 95,
                      "key": 179,
                    },
                    Object {
                      "doc_count": 34,
                      "key": 130,
                    },
                    Object {
                      "doc_count": 2,
                      "key": 177,
                    },
                    Object {
                      "doc_count": 2,
                      "key": 183,
                    },
                    Object {
                      "doc_count": 1,
                      "key": 174,
                    },
                  ],
                  "doc_count_error_upper_bound": 0,
                  "sum_other_doc_count": 3,
                },
                "Keywords": Object {
                  "buckets": Array [
                    Object {
                      "doc_count": 15,
                      "key": "Brca1",
                    },
                    Object {
                      "doc_count": 15,
                      "key": "Oncology",
                    },
                    Object {
                      "doc_count": 13,
                      "key": "Breast cancer",
                    },
                    Object {
                      "doc_count": 12,
                      "key": "Genetics & Heredity",
                    },
                    Object {
                      "doc_count": 10,
                      "key": "Biochemistry & Molecular Biology",
                    },
                  ],
                  "doc_count_error_upper_bound": 0,
                  "sum_other_doc_count": 641,
                },
              }
            }
            isMyDataSetPage={true}
            lookupFacetsList={
              Object {
                "Collection": "Collection (lookup)",
                "Subject": "Subject (lookup)",
              }
            }
            onFacetsChanged={[Function]}
            renameFacetsList={Object {}}
            showOpenAccessFilter={true}
          />
        </WithStyles(StandardRighthandCard)>
      </WithStyles(ForwardRef(Grid))>
    </Hidden>
  </WithStyles(ForwardRef(Grid))>
</StandardPage>
`;

exports[`myDatasets test renders loading screen while loading account data 1`] = `""`;

exports[`myDatasets test renders loading screen while loading publications  1`] = `
<StandardPage
  title="My research data"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(InlineLoader)
        message="Searching for your datasets"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</StandardPage>
`;

exports[`myDatasets test renders loading screen while loading publications while filtering 1`] = `
<StandardPage
  title="My research data"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={9}
      xs={12}
    >
      <StandardCard
        noHeader={true}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <span>
              Add to this list by 
              <a
                href="/data-collections/add"
              >
                adding a missing dataset
              </a>
              .
            </span>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <PublicationsListSorting
              bulkExportSize={1000}
              disabled={true}
              exportData={Object {}}
              initPageLength={20}
              onExportPublications={[Function]}
              onPageSizeChanged={[Function]}
              onSortByChanged={[Function]}
              pageSize={20}
              pagingData={Object {}}
              showDisplayAs={false}
              sortBy="published_date"
              sortDirection="Desc"
              sortingData={
                Object {
                  "bulkExportConfirmation": Object {
                    "confirmButtonLabel": "Close",
                    "confirmationMessage": <span
                      style={
                        Object {
                          "display": "block",
                          "margin": "1em 0",
                        }
                      }
                    >
                      Bulk export request has been queued. When the request has been processed, you will receive an email with the exported file as an attachment.
                    </span>,
                    "confirmationTitle": "Export queued successfully",
                  },
                  "displayRecordsAs": Array [
                    Object {
                      "index": 0,
                      "label": "Standard",
                      "value": "standard",
                    },
                    Object {
                      "index": 1,
                      "label": "Image Gallery",
                      "value": "image-gallery",
                    },
                  ],
                  "displayRecordsAsLabel": "Display results",
                  "exportOnlyLabel": "Export Only:",
                  "pageSize": "Works per page",
                  "recordsPerPage": Array [
                    10,
                    20,
                    50,
                    100,
                  ],
                  "sortBy": Array [
                    Object {
                      "label": "Published date",
                      "value": "published_date",
                    },
                    Object {
                      "label": "Search relevance",
                      "value": "score",
                    },
                    Object {
                      "label": "Title",
                      "value": "title",
                    },
                    Object {
                      "label": "Created date",
                      "value": "created_date",
                    },
                    Object {
                      "label": "Updated date",
                      "value": "updated_date",
                    },
                    Object {
                      "exclude": Array [
                        "image-gallery",
                      ],
                      "label": "Altmetric score",
                      "value": "altmetric_score",
                    },
                    Object {
                      "exclude": Array [
                        "image-gallery",
                      ],
                      "label": "Scopus citation count",
                      "value": "scopus_citation_count",
                    },
                    Object {
                      "exclude": Array [
                        "image-gallery",
                      ],
                      "label": "Web of Science citation count",
                      "value": "thomson_citation_count",
                    },
                  ],
                  "sortDirection": Array [
                    "Desc",
                    "Asc",
                  ],
                  "sortDirectionLabel": "Sort order",
                  "sortLabel": "Sort results by",
                }
              }
              sortingDefaults={Object {}}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={true}
              loading={true}
              onPageChanged={[Function]}
              pagingData={Object {}}
              pagingId="my-records-paging-top"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(ConfirmDialogBox)
              hideCancelButton={true}
              locale={
                Object {
                  "confirmButtonLabel": "Close",
                  "confirmationMessage": <span
                    style={
                      Object {
                        "display": "block",
                        "margin": "1em 0",
                      }
                    }
                  >
                    Bulk export request has been queued. When the request has been processed, you will receive an email with the exported file as an attachment.
                  </span>,
                  "confirmationTitle": "Export queued successfully",
                }
              }
              onRef={[Function]}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <div
              className="is-centered"
            >
              <WithStyles(InlineLoader)
                message="Retrieving your datasets"
              />
            </div>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={true}
              loading={true}
              onPageChanged={[Function]}
              pagingData={Object {}}
              pagingId="my-records-paging-bottom"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </StandardCard>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</StandardPage>
`;

exports[`myDatasets test renders no results 1`] = `
<StandardPage
  title="My research data"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <StandardCard
        text={
          <div>
            We were unable to find any results. You may be able to
             
            <a
              href="/data-collections/add"
            >
              add a missing dataset
            </a>
            .
          </div>
        }
        title="No datasets found"
      >
        <div>
          We were unable to find any results. You may be able to
           
          <a
            href="/data-collections/add"
          >
            add a missing dataset
          </a>
          .
        </div>
      </StandardCard>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</StandardPage>
`;
